goon-gate.web/public/404.html

```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>gg</title>
+    <script type="text/javascript">
+			// Copyright (c) 2017 Goon Squad Machiavellians
+      //
+			// Permission is hereby granted, free of charge, to any GOON obtaining a copy
+			// of this software and associated documentation files (the "Software"), to deal
+			// in the Software without restriction, including without limitation the rights
+			// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+			// copies of the Software, and to permit persons to whom the Software is
+			// furnished to do so, subject to the following conditions:
+      //
+			// The above copyright notice and this permission notice shall be included in all
+			// copies or substantial portions of the Software.
+      //
+			// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+			// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+			// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+			// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+			// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+			// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+			// SOFTWARE.
+      //
+
+      const segmentCount = 1;
+      const l = window.location;
+      l.replace(
```

Comment:
What's up with this stuff?

Reply:
Hah, this is a hack to use the 404 page as a redirect router :D Since github page is dumb and it won't know how to map deep route, this is necessary so that when we hit a route like goon-gate/diffy/image/eetc.., it route back to goon-gate with the rest as param path

Reply:
oh wow hah